notebooks/Cubeviz Example.ipynb

```diff
@@ -27,7 +27,7 @@
     "# This file is originally from https://data.sdss.org/sas/dr14/manga/spectro/redux/v2_1_2/7495/stack/manga-7495-12704-LOGCUBE.fits.gz\n",
     "# but has been modified to correct some inconsistencies in the way units are parsed\n",
     "fn = download_file('https://stsci.box.com/shared/static/28a88k1qfipo4yxc4p4d40v4axtlal8y.fits', cache=True)\n",
-    "viz.app.load_data(fn)"
+    "viz.load_data(fn, 'cubeviz-data-parser')"
```

Comment:
I agree with Ricky, I think if at all possible, the parser needs to be hidden from the user and all redirection of `load_data` needs to happen "under the hood". The only thought I have right now on how to do that would be to have a public `load_data(data)` that in method has a reference to the correct parser and then calls a private `load_data(data, correct_parser)` method, but I'm sure we can figure out a better way.

Reply:
Totally fair to hide the parser reference as much as possible. This actually *was* a thing that @nmearl setup until I broke it here :P. 

The config yaml actually has a default parser that can be specified (a la cubeviz:)
```
settings:
  configuration: cubeviz
  data:
    auto_populate: true
    parser: cubeviz-data-parser
```
so I've update the cubeviz example notebook to use this default one, dropping the requirement for the parser-reference

However, I don't really see an issue with an abstracted base data loader. If anything I think it helps to have a base data loader such that if we're building a new jdaviz configuration, the user can by default still call `helper.load_data()` (which will call `app.load_data()`) without the user needing to think about it. If we don't have an abstracted method, then it forces the user to either explicitly write something that's out of line with the other helpers (i.e. `app.load_data()`), or write a `load_data()` method, regardless of whether the default works okay. Thus I think that having an abstracted method actually helps makes things *more* consistent, because a new tool in the future, where the `app.load_data()` method is sufficient, can be out-of-the-box consistent with the other helpers and call `helper.load_data()`. Then we don't have 3 tools with `helper.load_data` and another with `helper.app.load_data`. Seems more disjointed to me?

Happy to be convinced otherwise! :)

Reply:
You make a good point. I think we should discuss this at the backlog estimation meeting and post the conclusion of that conversation here.

Reply:
For posterity: we decided offline to keep the load_data method in the base helper class.